hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieRealtimeInputFormatUtils.java

```diff
@@ -86,7 +86,7 @@ private static Configuration addProjectionField(Configuration conf, String field
 
   public static void addProjectionField(Configuration conf, String[] fieldName) {
     if (fieldName.length > 0) {
-      List<String> columnNameList = Arrays.stream(conf.get(serdeConstants.LIST_COLUMNS).split(",")).collect(Collectors.toList());
+      List<String> columnNameList = Arrays.stream(conf.get(serdeConstants.LIST_COLUMNS, "").split(",")).collect(Collectors.toList());
       Arrays.stream(fieldName).forEach(field -> {
```

Comment:
When the `LIST_COLUMNS` and when not?

Reply:
`LIST_COLUMNS` will be used when columns list is passed in Job Configuration & it wont be used in cases where Configuration is created with empty params such as `val jobConf = new JobConf()`  ( this is what we are doing currently in our Flink job to read a hoodie table) , due to this when `conf.get(serdeConstants.LIST_COLUMNS)` is invoked, it returns NPE so this particular fix will handle such cases